### ISSUE TITLE
Condition coverage limitation

### ISSUE THREAD

**Issue by JeanRochCoulon** [openhwgroup/cva6#1902]
### Is there an existing CVA6 bug for this?

- [X] I have searched the existing bug issues

### Bug Description

In the frontend.sv, the line 373 does not appear in the condition coverage report. To me, this is a condition coverage case. I am concerned because I think it hides an implementation bug. How to make it appear in the coverage report?

![image](https://github.com/openhwgroup/cva6/assets/56552370/858e6fb0-b2d8-414f-8810-054419f9585e)

@AyoubJalali @ASintzoff @AEzzejjari

**Comment by AyoubJalali** [comment]
Hello @JeanRochCoulon normally VCS can detect "?" condition by default, according to synopsys coverage manual :
![image](https://github.com/openhwgroup/cva6/assets/110232072/f865dcc9-ebca-45cf-9bb7-241156135dfb)

also branch coverage don't detect this of course.
So what a thing we should re-wrote the line to be visible for condition coverage

**Comment by JeanRochCoulon** [comment]
In the code, we have    **a = b ? c : d ;**    and   **not a <= b ? c : d ;**
That's maybe why it is not supported

**Comment by AnouarZajni** [comment]
Checked some condition coverage options but can't make the coverage, I think because it's a limitation while mixing "+" operand and expression. 
To make it work we need to split to 2 lines

**Comment by AnouarZajni** [comment]
Finaly, by splitting the line as following without any other condition coverage options the expression is covered:
![image](https://github.com/openhwgroup/cva6/assets/130553710/45ae7e2d-70a6-4050-bd14-a1c243ddb966)

**Comment by AyoubJalali** [comment]
@AnouarZajni Great !!
But this is only an example, should we change in the hole design for that ? I don't think so

**Comment by AnouarZajni** [comment]
Yes I can see many similar expression non-covered in design. You can find a non-exhaustive list in /shares/teams/riscv/fromAZ/missed_exp.txt
@JeanRochCoulon , can you please review the list?

**Comment by JeanRochCoulon** [comment]
Sorry for the long delay, but this inpu tis very interesting. To be discussed.

**Comment by AnouarZajni** [comment]
After some researchs, to enable such coverage we need to add "-cm_cond arith" in Makefile
@AyoubJalali will take care to update it. Waiting for the related PR

**Comment by AnouarZajni** [comment]
Waiting for PR #2694 to close. 
Thanks @AyoubJalali
